<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • farrisdata</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">farrisdata</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.11.900</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="http://github.com/jmw86069/farrisdata/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="http://github.com/jmw86069/farrisdata/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.11.900" id="version-0011900">version 0.0.11.900<a class="anchor" aria-label="anchor" href="#version-0011900"></a></h2>
<div class="section level3">
<h3 id="changes-0-0-11-900">changes<a class="anchor" aria-label="anchor" href="#changes-0-0-11-900"></a></h3>
<ul><li>
<code>farris_sashimi_files_df</code> was changed to update the <code>"url"</code> values to use a new server name <code>snpinfo.niehs.nih.gov</code> due to an obscure issue with web server certificate verification. The snpinfo certificate authority is more globally recognized, without the need to add certificate authorities.</li>
<li>Minor updates to data object help docs.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.10.900" id="version-0010900">version 0.0.10.900<a class="anchor" aria-label="anchor" href="#version-0010900"></a></h2>
<div class="section level3">
<h3 id="changes-0-0-10-900">changes<a class="anchor" aria-label="anchor" href="#changes-0-0-10-900"></a></h3>
<ul><li>Added <code>EnrichmentDFL</code> which is a list of 12 <code>data.frame</code> objects which each contain gene set enrichment stats results.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.9.900" id="version-009900">version 0.0.9.900<a class="anchor" aria-label="anchor" href="#version-009900"></a></h2>
<div class="section level3">
<h3 id="changes-0-0-9-900">changes<a class="anchor" aria-label="anchor" href="#changes-0-0-9-900"></a></h3>
<ul><li>
<code>farris_sashimi_files_df</code> was updated to use more accurate <code>scale_factor</code> values, calculated using housekeepers detected from Salmon gene quantification <code>farrisGenesSE</code> and re-calculated using coverage of each gene exon. For junction files, the junction counts annotated to each housekeeper gene were summed and used for normalization. Intermediate data was reviewed using MA-plots created with the “jamma” package on Github: <code>"jmw86069/jamma"</code>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.8.900" id="version-008900">version 0.0.8.900<a class="anchor" aria-label="anchor" href="#version-008900"></a></h2>
<div class="section level3">
<h3 id="changes-0-0-8-900">changes<a class="anchor" aria-label="anchor" href="#changes-0-0-8-900"></a></h3>
<ul><li>Updated bigWig urls to reference new union coverage files, produced using UCSC bigWigMerge, improved over previous data using bwtool.</li>
<li>Updated splice junction urls to refer to .gz compressed files, to improve import speed.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.7.900" id="version-007900">version 0.0.7.900<a class="anchor" aria-label="anchor" href="#version-007900"></a></h2>
<div class="section level3">
<h3 id="changes-0-0-7-900">changes<a class="anchor" aria-label="anchor" href="#changes-0-0-7-900"></a></h3>
<ul><li>Changed <code>farris_sashimi_files_df</code> to use union bigWig coverage files to reduce data bandwidth. Added <code>scale_factor</code> values for each coverage file and associated splice junction files. Updated the example docs to show exactly how the files_df data.frame was created.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.6.900" id="version-006900">version 0.0.6.900<a class="anchor" aria-label="anchor" href="#version-006900"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-0-0-6-900">bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-0-6-900"></a></h3>
<ul><li>Fixed issue with <code>farris_sashimi_files_df</code> that generated some URL entries which do not exist.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.5.900" id="version-005900">version 0.0.5.900<a class="anchor" aria-label="anchor" href="#version-005900"></a></h2>
<div class="section level3">
<h3 id="additions-0-0-5-900">additions<a class="anchor" aria-label="anchor" href="#additions-0-0-5-900"></a></h3>
<ul><li>
<code>farris_sashimi_files_df</code> data.frame was added to support <code><a href="https://rdrr.io/pkg/splicejam/man/prepareSashimi.html" class="external-link">splicejam::prepareSashimi()</a></code> which uses a data.frame to configure the various input locations for coverage and splice junction files.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.4.900" id="version-004900">version 0.0.4.900<a class="anchor" aria-label="anchor" href="#version-004900"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-0-0-4-900">bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-0-4-900"></a></h3>
<ul><li>Added “transcript_id” to colnames(rowData(farrisTxSE))</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.3.900" id="version-003900">version 0.0.3.900<a class="anchor" aria-label="anchor" href="#version-003900"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-0-0-3-900">bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-0-3-900"></a></h3>
<ul><li>Resolved a small issue in handling negative expression values for normalized TPM data, by adding +1 to all non-zero values. This adjustment ensures that all non-zero TPM values are still positive after normalization, while all values which were originally zero are maintained at zero. Note the TPM thresholds are already adjusted in the jampack Rmarkdown analysis workflow, to accomodate this adjustment.</li>
</ul></div>
<div class="section level3">
<h3 id="enhancements-0-0-3-900">enhancements<a class="anchor" aria-label="anchor" href="#enhancements-0-0-3-900"></a></h3>
<ul><li>Full docs were generated using pkgdown, describing each data object.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.2.900" id="version-002900">version 0.0.2.900<a class="anchor" aria-label="anchor" href="#version-002900"></a></h2>
<div class="section level3">
<h3 id="new-data-0-0-2-900">New data<a class="anchor" aria-label="anchor" href="#new-data-0-0-2-900"></a></h3>
<ul><li>
<code>GencodeVM12mm10cai</code> is a numeric vector of codon adaptability index (CAI) values for mouse mm10 Gencode vM12.</li>
<li>
<code>GencodeVM12mm10caiCtLow</code> is an integer count of codons where the CAI value is 0.5 or lower.</li>
<li>
<code>GencodeVM12mm10caiCtBad</code> is an integer count of codons where the CAI value is 0.3 or lower.</li>
<li>
<code>GencodeVM12mm10cai</code> is a numeric vector of codon adaptability index (CAI) values for mouse mm10 Gencode vM12.</li>
<li>
<code>GencodeVM12mm10cdsLength</code> is a numeric vector of CDS lengths for mouse mm10 Gencode vM12.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.1.900" id="version-001900">version 0.0.1.900<a class="anchor" aria-label="anchor" href="#version-001900"></a></h2>
<div class="section level3">
<h3 id="new-data-0-0-1-900">New data<a class="anchor" aria-label="anchor" href="#new-data-0-0-1-900"></a></h3>
<ul><li>AinsleyGenes, NakayamaGenes, CajigasGenes were added as reference data from three published studies detecting hippocampal CA1 dendrite genes, from mouse, mouse, and rat, respectively. These objects are character vectors of gene symbols.</li>
</ul></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by James M. Ward, Shannon Farris.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

