<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Farris et al transcript-level expression data — farrisTxSE • farrisdata</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Farris et al transcript-level expression data — farrisTxSE" />

<meta property="og:description" content="Farris et al transcript-level expression data" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">farrisdata</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.3.900</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jmw86069/farrisdata">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Farris et al transcript-level expression data</h1>
    <small class="dont-index">Source: <a href='https://github.com/jmw86069/farrisdata/blob/master/R/farrisdata.r'><code>R/farrisdata.r</code></a></small>
    <div class="hidden name"><code>farrisTxSE.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Farris et al transcript-level expression data</p>
    
    </div>

    <pre class="usage"><span class='no'>farrisTxSE</span></pre>
        
    <h2 class="hasAnchor" id="format"><a class="anchor" href="#format"></a>Format</h2>

    <p>SummarizedExperiment object with 122733 transcript rows, and 24
sample columns. Sample annotations are accessed as described
for <code><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>SummarizedExperiment-class</a></code>
objects, using <code><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>SummarizedExperiment::colData()</a></code>. Transcript and gene
annotations are accessed using <code><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>SummarizedExperiment::rowData()</a></code>.</p>
<p>The colData colnames are described as follows:</p>
<dl class='dl-horizontal'>
<dt>CellType</dt><dd><p>The major excitatory pyramidal cell types in
the mouse hippocampus, CA1, CA2, CA3, and dentate gyrus (DG).</p></dd>
<dt>Compartment</dt><dd><p>The cellular compartment within each
cell type: CB (cell body) is the pyramidal cell layer;
DE (dendrite) is stratum radiatum of CA1, CA2, CA3, and the
molecular layer of DG.</p></dd>
<dt>AnimalID</dt><dd><p>The number assigned to each mouse.</p></dd>
<dt>groupName</dt><dd><p>The sample group name associated with each
<code>CellType</code> and <code>Compartment</code>.</p></dd>
</dl>
    <p>The count data is accessed through the function <code>assays</code>
and has the following entries:</p>
<dl class='dl-horizontal'>
<dt>tpm</dt><dd><p>Median-normalized log2-transformed TPM values
generated by Salmon, after import using
<code><a href='https://www.rdocumentation.org/packages/tximport/topics/tximport'>tximport::tximport()</a></code>. The log2 transformation was
log2(1+x).</p></dd>
<dt>counts</dt><dd><p>Median-normalized log2-transformed pseudocount
data per-gene generated by Salmon, after import using
<code><a href='https://www.rdocumentation.org/packages/tximport/topics/tximport'>tximport::tximport()</a></code>. The log2 transformation was
log2(1+x).</p></dd>
<dt>raw_tpm</dt><dd><p>Raw log2-transformed TPM values
generated by Salmon, after import using
<code><a href='https://www.rdocumentation.org/packages/tximport/topics/tximport'>tximport::tximport()</a></code>. The log2 transformation was
log2(1+x).</p></dd>
<dt>raw_counts</dt><dd><p>The raw log2-transformed pseudocount data
per-gene generated by Salmon. The log2 transformation was
log2(1+x).</p></dd>
</dl>
    <p>The rowData contains gene annotations, which includes the relationship
between transcript and gene.</p>
<dl class='dl-horizontal'>
<dt>transcript_id</dt><dd><p>Gencode transcript_id as defined in the GTF file.</p></dd>
<dt>geneSymbol</dt><dd><p>Most current Entrez gene symbol associated with the
transcript_id, as of June 2018.</p></dd>
<dt>gene_type,transcript_type</dt><dd><p>Gencode attributes as defined in
the GTF file.</p></dd>
<dt>has3UTR,TxHas3UTR</dt><dd><p>Boolean indicating whether the gene or
transcript contained an annotated 3'UTR region defined in the
Gencode GTF file.</p></dd>
<dt>TxHasExt3UTR,GeneHasExt3UTR</dt><dd><p>Following 3'UTR extension
based upon RNA-seq data coverage, this boolean field indicates whether
each transcript or gene contains a 3'UTR with or without extension
by observed RNA-seq coverage.</p></dd>
<dt>TxDetectedByTPM</dt><dd><p>Boolean indicating whether each transcript
was considered "detected" based upon TPM and count data per gene,
as described in <code><a href='https://www.rdocumentation.org/packages/splicejam/topics/defineDetectedTx'>splicejam::defineDetectedTx()</a></code>.</p></dd>
</dl>
    
    <h2 class="hasAnchor" id="source"><a class="anchor" href="#source"></a>Source</h2>

    <p>Farris et al publication.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>SummarizedExperiment object containing transcript-level expression
data following Salmon quantitation, and import using
<code><a href='https://www.rdocumentation.org/packages/tximport/topics/tximport'>tximport::tximport()</a></code> at the transcript level.
Data is stored as log2-transformed TPM, and log2-transformed
pseudocounts from Salmon, after median normalization.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>farrisTxSE</span>);
<span class='co'>## Transcript-Gene annotation DataFrame</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>head</a></span>(<span class='kw pkg'>SummarizedExperiment</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>rowData</a></span>(<span class='no'>farrisTxSE</span>))</div><div class='output co'>#&gt; DataFrame with 6 rows and 10 columns
#&gt;                             transcript_id    geneSymbol            gene_type
#&gt;                               &lt;character&gt;   &lt;character&gt;          &lt;character&gt;
#&gt; ENSMUST00000193812.1 ENSMUST00000193812.1 4933401J01Rik                  TEC
#&gt; ENSMUST00000082908.1 ENSMUST00000082908.1       Gm26206                snRNA
#&gt; ENSMUST00000162897.1 ENSMUST00000162897.1          Xkr4       protein_coding
#&gt; ENSMUST00000159265.1 ENSMUST00000159265.1          Xkr4       protein_coding
#&gt; ENSMUST00000070533.4 ENSMUST00000070533.4          Xkr4       protein_coding
#&gt; ENSMUST00000192857.1 ENSMUST00000192857.1       Gm18956 processed_pseudogene
#&gt;                           transcript_type   has3UTR TxHas3UTR TxHasExt3UTR
#&gt;                               &lt;character&gt; &lt;logical&gt; &lt;logical&gt;    &lt;logical&gt;
#&gt; ENSMUST00000193812.1                  TEC     FALSE     FALSE        FALSE
#&gt; ENSMUST00000082908.1                snRNA     FALSE     FALSE        FALSE
#&gt; ENSMUST00000162897.1 processed_transcript      TRUE     FALSE        FALSE
#&gt; ENSMUST00000159265.1 processed_transcript      TRUE     FALSE        FALSE
#&gt; ENSMUST00000070533.4       protein_coding      TRUE      TRUE         TRUE
#&gt; ENSMUST00000192857.1 processed_pseudogene     FALSE     FALSE        FALSE
#&gt;                      GeneHasExt3UTR  TxHasCDS TxDetectedByTPM
#&gt;                           &lt;logical&gt; &lt;logical&gt;       &lt;logical&gt;
#&gt; ENSMUST00000193812.1          FALSE     FALSE           FALSE
#&gt; ENSMUST00000082908.1          FALSE     FALSE           FALSE
#&gt; ENSMUST00000162897.1           TRUE     FALSE            TRUE
#&gt; ENSMUST00000159265.1           TRUE     FALSE            TRUE
#&gt; ENSMUST00000070533.4           TRUE      TRUE            TRUE
#&gt; ENSMUST00000192857.1          FALSE     FALSE           FALSE</div><div class='input'>
<span class='co'>## Sample annotation DataFrame</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>head</a></span>(<span class='kw pkg'>SummarizedExperiment</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>colData</a></span>(<span class='no'>farrisTxSE</span>))</div><div class='output co'>#&gt; DataFrame with 6 rows and 4 columns
#&gt;          CellType   Region    AnimalID groupName
#&gt;          &lt;factor&gt; &lt;factor&gt; &lt;character&gt;  &lt;factor&gt;
#&gt; CA1CB492      CA1       CB         492    CA1_CB
#&gt; CA1CB496      CA1       CB         496    CA1_CB
#&gt; CA1CB502      CA1       CB         502    CA1_CB
#&gt; CA1DE492      CA1       DE         492    CA1_DE
#&gt; CA1DE496      CA1       DE         496    CA1_DE
#&gt; CA1DE502      CA1       DE         502    CA1_DE</div><div class='input'>
<span class='co'>## Assay data matrix for normalized TPM</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>head</a></span>(<span class='kw pkg'>SummarizedExperiment</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>assays</a></span>(<span class='no'>farrisTxSE</span>)<span class='kw'>[[</span><span class='st'>"tpm"</span>]]);</div><div class='output co'>#&gt;                      CA1CB492  CA1CB496 CA1CB502 CA1DE492 CA1DE496 CA1DE502
#&gt; ENSMUST00000193812.1 1.610332 0.7904088 1.158373 1.792008 1.311585 1.190829
#&gt; ENSMUST00000082908.1 0.000000 0.0000000 0.000000 0.000000 0.000000 0.000000
#&gt; ENSMUST00000162897.1 5.546307 5.4100021 5.704815 4.224328 5.491877 4.608420
#&gt; ENSMUST00000159265.1 4.900158 4.8581246 5.128569 4.785710 5.218501 3.267856
#&gt; ENSMUST00000070533.4 4.819636 4.9697168 4.962035 3.960422 4.062039 3.999063
#&gt; ENSMUST00000192857.1 1.586775 1.3612144 1.715048 0.000000 0.000000 0.000000
#&gt;                      CA2CB492 CA2CB496 CA2CB502 CA2DE492 CA2DE496 CA2DE502
#&gt; ENSMUST00000193812.1 1.356880 1.122915 1.922335 1.548804 1.163878 0.000000
#&gt; ENSMUST00000082908.1 0.000000 0.000000 0.000000 0.000000 0.000000 0.000000
#&gt; ENSMUST00000162897.1 4.882472 5.319941 4.528681 3.287487 3.987989 4.152704
#&gt; ENSMUST00000159265.1 4.534462 4.779240 4.552718 4.084830 3.873951 3.859247
#&gt; ENSMUST00000070533.4 4.804406 4.889756 4.254999 3.394788 3.973354 3.841563
#&gt; ENSMUST00000192857.1 0.000000 1.056060 0.000000 0.000000 0.000000 0.000000
#&gt;                      CA3CB492  CA3CB496  CA3CB502 CA3DE492 CA3DE496 CA3DE502
#&gt; ENSMUST00000193812.1 0.000000 0.9701784 0.9787647 0.000000 1.301224 0.000000
#&gt; ENSMUST00000082908.1 0.000000 0.0000000 0.0000000 0.000000 0.000000 0.000000
#&gt; ENSMUST00000162897.1 5.646500 5.7608985 5.6991368 4.682759 5.313276 4.654305
#&gt; ENSMUST00000159265.1 5.506492 4.8380664 5.4546917 3.988007 4.436601 4.002717
#&gt; ENSMUST00000070533.4 5.310857 5.2492192 5.3960733 4.334719 4.413020 4.157208
#&gt; ENSMUST00000192857.1 0.000000 1.5506465 0.0000000 0.000000 0.000000 0.000000
#&gt;                        DGCB492   DGCB496  DGCB502   DGDE492  DGDE496  DGDE502
#&gt; ENSMUST00000193812.1 0.4902402 0.6670767 0.643377 0.9595546 0.000000 1.044818
#&gt; ENSMUST00000082908.1 0.0000000 0.0000000 0.000000 0.0000000 0.000000 0.000000
#&gt; ENSMUST00000162897.1 4.1109933 4.5889056 4.471882 3.5582606 3.565559 3.518744
#&gt; ENSMUST00000159265.1 4.4897173 4.0607367 3.978226 2.6384302 2.929368 2.927404
#&gt; ENSMUST00000070533.4 3.9907276 4.0903952 3.950510 3.2993651 3.718393 3.322432
#&gt; ENSMUST00000192857.1 0.0000000 1.2529288 1.871817 0.0000000 0.000000 0.000000</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      
      <li><a href="#format">Format</a></li>

      <li><a href="#source">Source</a></li>

      <li><a href="#details">Details</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by James M. Ward, Shannon Farris.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

