<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Farris et al transcript-level expression data — farrisTxSE • farrisdata</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Farris et al transcript-level expression data — farrisTxSE" />

<meta property="og:description" content="Farris et al transcript-level expression data" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">farrisdata</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.9.900</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jmw86069/farrisdata">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Farris et al transcript-level expression data</h1>
    <small class="dont-index">Source: <a href='https://github.com/jmw86069/farrisdata/blob/master/R/farrisdata.r'><code>R/farrisdata.r</code></a></small>
    <div class="hidden name"><code>farrisTxSE.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Farris et al transcript-level expression data</p>
    
    </div>

    <pre class="usage"><span class='no'>farrisTxSE</span></pre>
        
    <h2 class="hasAnchor" id="format"><a class="anchor" href="#format"></a>Format</h2>

    <p>SummarizedExperiment object with 122733 transcript rows, and 24
sample columns. Sample annotations are accessed as described
for <code><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>SummarizedExperiment-class</a></code>
objects, using <code><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>SummarizedExperiment::colData()</a></code>. Transcript and gene
annotations are accessed using <code><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>SummarizedExperiment::rowData()</a></code>.</p>
<p>The colData colnames are described as follows:</p>
<dl class='dl-horizontal'>
<dt>CellType</dt><dd><p>The major excitatory pyramidal cell types in
the mouse hippocampus, CA1, CA2, CA3, and dentate gyrus (DG).</p></dd>
<dt>Compartment</dt><dd><p>The cellular compartment within each
cell type: CB (cell body) is the pyramidal cell layer;
DE (dendrite) is stratum radiatum of CA1, CA2, CA3, and the
molecular layer of DG.</p></dd>
<dt>AnimalID</dt><dd><p>The number assigned to each mouse.</p></dd>
<dt>groupName</dt><dd><p>The sample group name associated with each
<code>CellType</code> and <code>Compartment</code>.</p></dd>
</dl>
    <p>The count data is accessed through the function <code>assays</code>
and has the following entries:</p>
<dl class='dl-horizontal'>
<dt>tpm</dt><dd><p>Median-normalized log2-transformed TPM values
generated by Salmon, after import using
<code><a href='https://www.rdocumentation.org/packages/tximport/topics/tximport'>tximport::tximport()</a></code>. The log2 transformation was
log2(1+x).</p></dd>
<dt>counts</dt><dd><p>Median-normalized log2-transformed pseudocount
data per-gene generated by Salmon, after import using
<code><a href='https://www.rdocumentation.org/packages/tximport/topics/tximport'>tximport::tximport()</a></code>. The log2 transformation was
log2(1+x).</p></dd>
<dt>raw_tpm</dt><dd><p>Raw log2-transformed TPM values
generated by Salmon, after import using
<code><a href='https://www.rdocumentation.org/packages/tximport/topics/tximport'>tximport::tximport()</a></code>. The log2 transformation was
log2(1+x).</p></dd>
<dt>raw_counts</dt><dd><p>The raw log2-transformed pseudocount data
per-gene generated by Salmon. The log2 transformation was
log2(1+x).</p></dd>
</dl>
    <p>The rowData contains gene annotations, which includes the relationship
between transcript and gene.</p>
<dl class='dl-horizontal'>
<dt>transcript_id</dt><dd><p>Gencode transcript_id as defined in the GTF file.</p></dd>
<dt>geneSymbol</dt><dd><p>Most current Entrez gene symbol associated with the
transcript_id, as of June 2018.</p></dd>
<dt>gene_type,transcript_type</dt><dd><p>Gencode attributes as defined in
the GTF file.</p></dd>
<dt>has3UTR,TxHas3UTR</dt><dd><p>Boolean indicating whether the gene or
transcript contained an annotated 3'UTR region defined in the
Gencode GTF file.</p></dd>
<dt>TxHasExt3UTR,GeneHasExt3UTR</dt><dd><p>Following 3'UTR extension
based upon RNA-seq data coverage, this boolean field indicates whether
each transcript or gene contains a 3'UTR with or without extension
by observed RNA-seq coverage.</p></dd>
<dt>TxDetectedByTPM</dt><dd><p>Boolean indicating whether each transcript
was considered "detected" based upon TPM and count data per gene,
as described in <code><a href='https://www.rdocumentation.org/packages/splicejam/topics/defineDetectedTx'>splicejam::defineDetectedTx()</a></code>.</p></dd>
</dl>
    
    <h2 class="hasAnchor" id="source"><a class="anchor" href="#source"></a>Source</h2>

    <p>Farris et al publication.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>SummarizedExperiment object containing transcript-level expression
data following Salmon quantitation, and import using
<code><a href='https://www.rdocumentation.org/packages/tximport/topics/tximport'>tximport::tximport()</a></code> at the transcript level.
Data is stored as log2-transformed TPM, and log2-transformed
pseudocounts from Salmon, after median normalization.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other farrisdata SummarizedExperiment data: <code><a href='farrisGeneSE.html'>farrisGeneSE</a></code></p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>farrisTxSE</span>);
<span class='co'>## Transcript-Gene annotation DataFrame</span>
<span class='kw'>if</span> (<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/message'>suppressPackageStartupMessages</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/library'>require</a></span>(<span class='no'>knitr</span>))) {
   <span class='kw pkg'>knitr</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/knitr/topics/kable'>kable</a></span>(
      <span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>tail</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>head</a></span>(<span class='kw pkg'>SummarizedExperiment</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>rowData</a></span>(<span class='no'>farrisTxSE</span>), <span class='fl'>15385</span>), <span class='fl'>10</span>)
   )
} <span class='kw'>else</span> {
   <span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>tail</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>head</a></span>(<span class='kw pkg'>SummarizedExperiment</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>rowData</a></span>(<span class='no'>farrisTxSE</span>), <span class='fl'>15385</span>), <span class='fl'>10</span>)
}</div><div class='output co'>#&gt; 
#&gt; 
#&gt; |                      |transcript_id         |geneSymbol |gene_type            |transcript_type      |has3UTR |TxHas3UTR |TxHasExt3UTR |GeneHasExt3UTR |TxHasCDS |TxDetectedByTPM |
#&gt; |:---------------------|:---------------------|:----------|:--------------------|:--------------------|:-------|:---------|:------------|:--------------|:--------|:---------------|
#&gt; |ENSMUST00000125292.1  |ENSMUST00000125292.1  |Gria1      |protein_coding       |protein_coding       |TRUE    |FALSE     |TRUE         |TRUE           |TRUE     |FALSE           |
#&gt; |ENSMUST00000036315.15 |ENSMUST00000036315.15 |Gria1      |protein_coding       |protein_coding       |TRUE    |TRUE      |TRUE         |TRUE           |TRUE     |TRUE            |
#&gt; |ENSMUST00000094179.10 |ENSMUST00000094179.10 |Gria1      |protein_coding       |protein_coding       |TRUE    |TRUE      |TRUE         |TRUE           |TRUE     |TRUE            |
#&gt; |ENSMUST00000151045.2  |ENSMUST00000151045.2  |Gria1      |protein_coding       |protein_coding       |TRUE    |TRUE      |TRUE         |TRUE           |TRUE     |FALSE           |
#&gt; |ENSMUST00000173531.1  |ENSMUST00000173531.1  |Gria1      |protein_coding       |processed_transcript |TRUE    |FALSE     |FALSE        |TRUE           |FALSE    |FALSE           |
#&gt; |ENSMUST00000118824.1  |ENSMUST00000118824.1  |Gm12242    |processed_pseudogene |processed_pseudogene |FALSE   |FALSE     |FALSE        |FALSE          |FALSE    |FALSE           |
#&gt; |ENSMUST00000020831.12 |ENSMUST00000020831.12 |Fam114a2   |protein_coding       |protein_coding       |TRUE    |TRUE      |TRUE         |TRUE           |TRUE     |TRUE            |
#&gt; |ENSMUST00000108850.1  |ENSMUST00000108850.1  |Fam114a2   |protein_coding       |protein_coding       |TRUE    |TRUE      |TRUE         |TRUE           |TRUE     |FALSE           |
#&gt; |ENSMUST00000129596.1  |ENSMUST00000129596.1  |Fam114a2   |protein_coding       |processed_transcript |TRUE    |FALSE     |FALSE        |TRUE           |FALSE    |FALSE           |
#&gt; |ENSMUST00000130596.1  |ENSMUST00000130596.1  |Fam114a2   |protein_coding       |processed_transcript |TRUE    |FALSE     |FALSE        |TRUE           |FALSE    |FALSE           |</div><div class='input'>
<span class='co'>## Sample annotation DataFrame</span>
<span class='kw'>if</span> (<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/message'>suppressPackageStartupMessages</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/library'>require</a></span>(<span class='no'>knitr</span>))) {
   <span class='kw pkg'>knitr</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/knitr/topics/kable'>kable</a></span>(
      <span class='kw pkg'>SummarizedExperiment</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>colData</a></span>(<span class='no'>farrisTxSE</span>)
   )
} <span class='kw'>else</span> {
   <span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>head</a></span>(<span class='kw pkg'>SummarizedExperiment</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>colData</a></span>(<span class='no'>farrisTxSE</span>));
}</div><div class='output co'>#&gt; 
#&gt; 
#&gt; |         |CellType |Region |AnimalID |groupName |
#&gt; |:--------|:--------|:------|:--------|:---------|
#&gt; |CA1CB492 |CA1      |CB     |492      |CA1_CB    |
#&gt; |CA1CB496 |CA1      |CB     |496      |CA1_CB    |
#&gt; |CA1CB502 |CA1      |CB     |502      |CA1_CB    |
#&gt; |CA1DE492 |CA1      |DE     |492      |CA1_DE    |
#&gt; |CA1DE496 |CA1      |DE     |496      |CA1_DE    |
#&gt; |CA1DE502 |CA1      |DE     |502      |CA1_DE    |
#&gt; |CA2CB492 |CA2      |CB     |492      |CA2_CB    |
#&gt; |CA2CB496 |CA2      |CB     |496      |CA2_CB    |
#&gt; |CA2CB502 |CA2      |CB     |502      |CA2_CB    |
#&gt; |CA2DE492 |CA2      |DE     |492      |CA2_DE    |
#&gt; |CA2DE496 |CA2      |DE     |496      |CA2_DE    |
#&gt; |CA2DE502 |CA2      |DE     |502      |CA2_DE    |
#&gt; |CA3CB492 |CA3      |CB     |492      |CA3_CB    |
#&gt; |CA3CB496 |CA3      |CB     |496      |CA3_CB    |
#&gt; |CA3CB502 |CA3      |CB     |502      |CA3_CB    |
#&gt; |CA3DE492 |CA3      |DE     |492      |CA3_DE    |
#&gt; |CA3DE496 |CA3      |DE     |496      |CA3_DE    |
#&gt; |CA3DE502 |CA3      |DE     |502      |CA3_DE    |
#&gt; |DGCB492  |DG       |CB     |492      |DG_CB     |
#&gt; |DGCB496  |DG       |CB     |496      |DG_CB     |
#&gt; |DGCB502  |DG       |CB     |502      |DG_CB     |
#&gt; |DGDE492  |DG       |DE     |492      |DG_DE     |
#&gt; |DGDE496  |DG       |DE     |496      |DG_DE     |
#&gt; |DGDE502  |DG       |DE     |502      |DG_DE     |</div><div class='input'>
<span class='co'>## Assay data matrix for normalized TPM</span>
<span class='kw'>if</span> (<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/message'>suppressPackageStartupMessages</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/library'>require</a></span>(<span class='no'>knitr</span>))) {
   <span class='kw pkg'>knitr</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/knitr/topics/kable'>kable</a></span>(
      <span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>tail</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>head</a></span>(<span class='kw pkg'>SummarizedExperiment</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>assays</a></span>(<span class='no'>farrisTxSE</span>)<span class='kw'>[[</span><span class='st'>"tpm"</span>]], <span class='fl'>15385</span>), <span class='fl'>10</span>)
   )
} <span class='kw'>else</span> {
   <span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>tail</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>head</a></span>(<span class='kw pkg'>SummarizedExperiment</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>assays</a></span>(<span class='no'>farrisTxSE</span>)<span class='kw'>[[</span><span class='st'>"tpm"</span>]], <span class='fl'>15385</span>), <span class='fl'>10</span>)
}</div><div class='output co'>#&gt; 
#&gt; 
#&gt; |                      |  CA1CB492|  CA1CB496|  CA1CB502| CA1DE492| CA1DE496| CA1DE502|  CA2CB492|  CA2CB496| CA2CB502| CA2DE492| CA2DE496| CA2DE502|  CA3CB492|  CA3CB496|  CA3CB502| CA3DE492| CA3DE496|  CA3DE502|   DGCB492|   DGCB496|   DGCB502|   DGDE492|  DGDE496|   DGDE502|
#&gt; |:---------------------|---------:|---------:|---------:|--------:|--------:|--------:|---------:|---------:|--------:|--------:|--------:|--------:|---------:|---------:|---------:|--------:|--------:|---------:|---------:|---------:|---------:|---------:|--------:|---------:|
#&gt; |ENSMUST00000125292.1  | 3.5237283| 3.2301712| 4.2309811| 1.764187| 1.277686| 1.182975| 1.2878658| 2.5620205| 1.241757| 3.265422| 1.036398| 1.106588| 3.5506856| 3.3010433| 3.0429398| 2.327770| 2.619280| 0.9253744| 1.5316583| 3.1453899| 3.4948050| 0.9594580| 1.103800| 0.9598450|
#&gt; |ENSMUST00000036315.15 | 7.3740560| 7.5963403| 7.7930511| 4.867291| 7.052011| 5.980195| 7.8868671| 7.8907566| 7.486656| 5.126341| 5.572148| 5.968372| 7.9780124| 8.0113010| 8.2236747| 6.389772| 6.523105| 6.2393871| 7.2841486| 6.9844536| 7.2228444| 5.2028549| 5.223238| 5.1002982|
#&gt; |ENSMUST00000094179.10 | 8.6282778| 8.7770955| 8.8152266| 6.268609| 6.542139| 6.835774| 6.3836848| 6.5871921| 6.331986| 4.927524| 5.984579| 5.772234| 5.7230742| 6.0346963| 5.8199319| 6.660683| 6.539273| 6.4534405| 7.8034890| 7.8193713| 7.7759722| 6.0482512| 5.725406| 5.9380773|
#&gt; |ENSMUST00000151045.2  | 3.1844520| 2.7963751| 2.0789046| 2.116279| 1.440427| 1.776318| 2.2040931| 2.9158568| 2.311911| 2.116180| 1.263960| 2.222698| 2.7082691| 3.3410402| 2.5303085| 1.676808| 1.428374| 1.3589163| 2.5154027| 2.2626368| 2.2975434| 1.8448737| 1.352283| 2.0238565|
#&gt; |ENSMUST00000173531.1  | 2.0097707| 1.2863568| 2.9025973| 1.766633| 1.278707| 1.185306| 1.0568596| 1.5785033| 1.245786| 1.549274| 2.725294| 1.109378| 0.9326488| 1.9065740| 2.7233531| 1.165464| 1.173444| 3.2217215| 2.8482967| 2.7714623| 2.0157174| 0.9610732| 2.689475| 0.9642246|
#&gt; |ENSMUST00000118824.1  | 0.7973897| 0.7891245| 0.7257397| 1.764282| 1.277701| 1.183009| 0.8569493| 0.9256989| 1.241768| 1.548128| 1.083246| 1.106601| 0.9279031| 0.8797852| 1.9205797| 1.161361| 1.171640| 1.0153860| 0.4866034| 0.7306812| 0.8296252| 0.9594608| 1.150628| 1.0196834|
#&gt; |ENSMUST00000020831.12 | 4.5692848| 4.4050921| 4.3853111| 2.903407| 3.445532| 3.546994| 4.1621070| 4.2100802| 4.308294| 4.015971| 4.050088| 3.955258| 3.9404952| 4.0778114| 3.8534383| 3.970655| 3.356445| 3.8518254| 5.2507571| 5.2429177| 5.2670502| 4.0743524| 4.443832| 4.2081596|
#&gt; |ENSMUST00000108850.1  | 0.8429474| 0.9013310| 0.8435403| 1.769204| 1.291345| 1.211227| 0.9215645| 1.0085776| 1.266666| 1.571436| 1.753920| 1.139032| 0.9955124| 1.0332869| 1.9094915| 1.245723| 1.186544| 1.2707682| 0.5726204| 0.9495726| 0.7595002| 1.0026786| 1.564715| 1.0411541|
#&gt; |ENSMUST00000129596.1  | 2.0056370| 0.7906439| 0.7259676| 0.000000| 0.000000| 1.183011| 0.8572991| 0.9265212| 1.242298| 1.549239| 0.000000| 1.106734| 0.9280149| 0.8803533| 0.8649923| 1.161455| 1.171677| 0.9254341| 1.2892238| 1.7415312| 1.3317189| 0.9594809| 1.103841| 0.9599639|
#&gt; |ENSMUST00000130596.1  | 0.8318545| 0.8001594| 0.7459478| 3.038650| 3.226690| 1.198608| 0.9254844| 0.9675445| 3.168013| 1.549761| 1.039774| 1.107790| 2.6613043| 0.9033893| 2.6675268| 4.347363| 1.179759| 1.3512509| 0.7118065| 1.9969306| 0.6572023| 1.0394576| 2.235069| 0.9663209|</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      
      <li><a href="#format">Format</a></li>

      <li><a href="#source">Source</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by James M. Ward, Shannon Farris.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

